File: 206-ReverseLinkedList/206-ReverseLinkedList.py
# Last updated: 7/24/2025, 8:33:01 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        prev=head
        curr=head.next
        prev.next=None

        while curr.next:
            succ = curr.next
            curr.next = prev
            prev = curr
            curr = succ
        curr.next=prev
        return curr
        

File: 206-ReverseLinkedList/206-ReverseLinkedList.cpp
// Last updated: 7/24/2025, 8:32:57 AM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ios::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        ListNode * prev = NULL;
        ListNode * node = head;

        while(node){
            cout<<node->val<<" ";
            ListNode *temp = node->next;
            node->next = prev;
            prev = node;
            node = temp;
        }
        return prev;
    }
};